tests/adapter/dbt/tests/adapter/test_incremental_unique_id.py

```diff
@@ -0,0 +1,584 @@
+        row_count_query = "select * from {}.{}".format(project.test_schema, seed)
+        project.run_sql_file(Path("seeds") / Path(update_sql_file + ".sql"))
+        seed_rows = len(project.run_sql(row_count_query, fetch="all"))
+
+        # propagate seed state to incremental model according to unique keys
+        inc_test_model_count = self.update_incremental_model(incremental_model=incremental_model)
+
+        return (seed_count, model_count, seed_rows, inc_test_model_count)
+
+    def check_scenario_correctness(self, expected_fields, test_case_fields, project):
+        """Invoke assertions to verify correct build functionality"""
+        # 1. test seed(s) should build afresh
+        assert expected_fields.seed_count == test_case_fields.seed_count
+        # 2. test model(s) should build afresh
+        assert expected_fields.model_count == test_case_fields.model_count
+        # 3. seeds should have intended row counts post update
+        assert expected_fields.seed_rows == test_case_fields.seed_rows
+        # 4. incremental test model(s) should be updated
+        assert expected_fields.inc_test_model_count == test_case_fields.inc_test_model_count
+        # 5. extra incremental model(s) should be built; optional since
+        #   comparison may be between an incremental model and seed
+        if expected_fields.opt_model_count and test_case_fields.opt_model_count:
+            assert expected_fields.opt_model_count == test_case_fields.opt_model_count
+        # 6. result table should match intended result set (itself a relation)
+        table_comp = TableComparison(
+            adapter=project.adapter, unique_schema=project.test_schema, database=project.database
+        )
+        table_comp.assert_tables_equal(expected_fields.relation, test_case_fields.relation)
+
+    def stub_expected_fields(self, relation, seed_rows, opt_model_count=None):
+        return ResultHolder(
+            seed_count=1,
+            model_count=1,
+            seed_rows=seed_rows,
+            inc_test_model_count=1,
+            opt_model_count=opt_model_count,
+            relation=relation,
+        )
+
+    def fail_to_build_inc_missing_unique_key_column(self, incremental_model_name):
+        """should pass back error state when trying build an incremental
+        model whose unique key or keylist includes a column missing
+        from the incremental model"""
+        seed_count = len(run_dbt(["seed", "--select", "seed", "--full-refresh"]))  # noqa:F841
+        # unique keys are not applied on first run, so two are needed
+        run_dbt(
+            ["run", "--select", incremental_model_name, "--full-refresh"],
+            expect_pass=True,
+        )
+        run_result = run_dbt(
+            ["run", "--select", incremental_model_name], expect_pass=False
+        ).results[0]
+
+        return run_result.status, run_result.message
+
+    # no unique_key test
+    def test__no_unique_keys(self, project):
```

Comment:
I think the factoring in the way these tests are done could be more straightforward and easier to read. I would recommend that you do something like:

```
  def test__no_unique_keys(self, project):
        """with no unique keys, seed and model should match"""

        expected_fields = self.get_expected_fields(relation="seed", seed_rows=8)
        test_case_fields = self.get_test_fields(
            project, seed="seed", incremental_model="no_unique_key", update_sql_file="add_new_rows"
        )
        self.check_scenario_correctness(project, expected_fields, test_case_fields)

```

Reply:
Rename stub_expected_fields to 'get_expected_fields'. Rename 'setup_test' to 'get_test_fields'. Refactor get_test_fields to return the ResultHolder object and change the signature to use keywords, except for project, which should be first.